### PR TITLE
Fix git-hg mirror tag sorting so jdk11u _adopt tags get created

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
The git-hg script needed the jdk tag sorting updating to allow for the jdk-11.0.9.1 (patch) scenario. As it was it thought jdk-11.0.9.1+1 was the latest tag, as opposed to jdk-11.0.10+8.

Sorting logic copied from: https://github.com/AdoptOpenJDK/openjdk-build/blob/4b03e4c4c37a5c02ba60caacb4bd2d3bd9f6fead/sbin/build.sh#L802

Signed-off-by: Andrew Leonard <anleonar@redhat.com>

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
Tested on fork here: https://github.com/andrew-m-leonard/openjdk-jdk11u
Test build from which here: https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-linux-ppc64le-hotspot/792/console

File: git-hg/mercurialToGit.sh [hunk 1]
```diff
@@ -74,19 +74,21 @@ fi
...
 #
 # First, sort on build number (B):
-jdk11plus_tag_sort1="sort -t+ -k2n"
-# Second, (stable) sort on (V), (W), (X):
-jdk11plus_tag_sort2="sort -t. -k2n -k3n -k4n -s"
-jdk11plus_sort_tags_cmd="grep -v _adopt | sed 's/jdk-/jdk./g' | $jdk11plus_tag_sort1 | $jdk11plus_tag_sort2 | sed 's/jdk./jdk-/g'"
+jdk11plus_tag_sort1="sort -t+ -k2,2n"
+# Second, (stable) sort on (V), (W), (X), (P): P(Patch) is optional and defaulted to "0"
+jdk11plus_tag_sort2="sort -t. -k2,2n -k3,3n -k4,4n -k5,5n"
+jdk11plus_sort_tags_cmd="grep -v _openj9 | grep -v _adopt | sed 's/jdk-/jdk./g' | sed 's/+/.0.+/g' | $jdk11plus_tag_sort1 | nl | $jdk11plus_tag_sort2 | sed 's/\.0\.+/+/g' | cut -f2- | sed 's/jdk./jdk-/g'"
```
Comment (karianna) [inline git-hg/mercurialToGit.sh:91]: is `cut -f2-` correct?

Comment (andrew-m-leonard) [reply git-hg/mercurialToGit.sh:91]: yes, this is copied from sbin/build.sh. Basically, to make the cmd work on Solaris that does not support "stable" sorting, we use the "trick" of line numbering ("nl") after the first sort, then doing sort2, then cut -f2- removes the first field "line number".

**Review by karianna — APPROVED** [review]
lgtm

**Review by gdams — APPROVED** [review]

**Pull request merged**